### ISSUE TITLE
Support for Snaking Columns?

### ISSUE THREAD

**Issue by SahandSoltanieh** [MP70/pdf-lib-draw-table#4]
Hi , I've been recently testing your table generator and I really liked it. I wanted to ask if there’s a feature for snaking columns—where the table flows across multiple vertical columns with repeated headers in each column? If it's not currently supported, is adding this feature on your roadmap? Thanks for your great work @MP70 !

**Comment by MP70** [comment]
Hi @imsahsol,

It's not natively supported in the sense that you could just add a param and it happens for you, but this could be achieved using this library. You would need to catch the "ERR_TABLE_HEIGHT_OVERFLOW" error code and then snake your data array as needed. This library is mainly expected to be quite low level.

To be honest adding native support is not a major priority for me as I wouldn't use it myself, but I'm happy to take a working PR if you're interested in contributing this feature!

**Comment by imsahsol** [comment]
Thanks for the quick response. I will explore your suggestion and consider working on this feature and submitting a PR if I manage to get it working well. If it's ok let's have this issue open for now (unless it became stale for a while) in case I faced issues that might be helpful to get your advice on.

**Comment by MP70** [comment]
No problem, let me know as and when if I can help :). Is this the effect you were looking for @imsahsol ?
Instead of this (using full width just for demo)
<img width="576" alt="Screenshot 2024-06-14 at 11 57 42" src="https://github.com/MP70/pdf-lib-draw-table/assets/29546576/8d48ac60-cef5-41aa-97fc-f971c3e669ee">
snake like this (snake once just for demo)
<img width="575" alt="Screenshot 2024-06-14 at 11 58 06" src="https://github.com/MP70/pdf-lib-draw-table/assets/29546576/b93212f8-94dd-4cea-a511-c40350f3b2c1">

**Comment by imsahsol** [comment]
> No problem, let me know as and when if I can help :). Is this the effect you were looking for @imsahsol ? Instead of this (using full width just for demo) <img alt="Screenshot 2024-06-14 at 11 57 42" width="576" src="https://private-user-images.githubusercontent.com/29546576/339741720-8d48ac60-cef5-41aa-97fc-f971c3e669ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzODE0NzcsIm5iZiI6MTcxODM4MTE3NywicGF0aCI6Ii8yOTU0NjU3Ni8zMzk3NDE3MjAtOGQ0OGFjNjAtY2VmNS00MWFhLTk3ZmMtZjk3MWMzZTY2OWVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE0VDE2MDYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiYWI0ZGIxOWVkZTBlZGNkOGEzZGNiMzIyYzk4YTk4ODE1M2U3OThlOWQ3ZTU4ZWJjZjZhZDg0MzNhNTViNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wHq7x53RYhj_Oxc683LzrgO8gvS1BsyF8rf10VQs4Fk"> snake like this (snake once just for demo) <img alt="Screenshot 2024-06-14 at 11 58 06" width="575" src="https://private-user-images.githubusercontent.com/29546576/339741827-b93212f8-94dd-4cea-a511-c40350f3b2c1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzODE0NzcsIm5iZiI6MTcxODM4MTE3NywicGF0aCI6Ii8yOTU0NjU3Ni8zMzk3NDE4MjctYjkzMjEyZjgtOTRkZC00Y2VhLWE1MTEtYzQwMzUwZjNiMmMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE0VDE2MDYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmOTM3ZGE3ZjgyZjI4MTQyYmM4YmMxNTNkNTlhYmE5OTA1NGFiYThjMTg2NmU3NDMxYTA3OTA3NTJmMTM4ZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kXf5i5iDn4NC1kTj083kooT45oAdFlcTbzohZgju-tA">

@MP70, that's pretty much it. Just so you know, there has been a similar long-standing issue in the `pdfmake` repository for quite some time now. Here's the link to that issue in case you're interested in learning more about this requested behavior: https://github.com/bpampuch/pdfmake/issues/604#issuecomment-1624578343

(BTW, your demo page looks really neat. Can I ask which libraries you used for that? Do you have a public repo for it if someone wants to check it out?)

**Comment by MP70** [comment]
Fair enough. Doubt this is too hard to get done properly, but a basic implementation can already trivially be done on the top of, rather than in the library (though a more thorough one could be slightly easier/more efficient inside). The "Snake Columns on Vertical Overflow" that I quickly popped in the playground is ~a dozen LOCs called on receipt of the overflow error, admittedly it's very basic though.

Dummy data for playground incase it's helpful.
[tableOptions.json](https://github.com/user-attachments/files/15841715/tableOptions.json)

**Comment by MP70** [comment]
> (BTW, your demo page looks really neat. Can I ask which libraries you used for that? Do you have a public repo for it if someone wants to check it out?)

Thank you, but actually this demo is very rough, I just threw it together to save myself time with manual/visual testing. Mainly since the code is quite rough (as it was mostly for my personal use) it's not open source at the moment.
